<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>748</width>
    <height>589</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>QT OpenGL</string>
  </property>
  <property name="layoutDirection">
   <enum>Qt::LeftToRight</enum>
  </property>
  <property name="autoFillBackground">
   <bool>false</bool>
  </property>
  <property name="iconSize">
   <size>
    <width>20</width>
    <height>19</height>
   </size>
  </property>
  <property name="toolButtonStyle">
   <enum>Qt::ToolButtonIconOnly</enum>
  </property>
  <property name="dockNestingEnabled">
   <bool>false</bool>
  </property>
  <widget class="QWidget" name="centralWidget">
   <layout class="QHBoxLayout" name="horizontalLayout">
    <item>
     <layout class="QVBoxLayout" name="verticalLayout">
      <item>
       <widget class="gl_widget" name="widget" native="true">
        <property name="minimumSize">
         <size>
          <width>300</width>
          <height>300</height>
         </size>
        </property>
       </widget>
      </item>
     </layout>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>748</width>
     <height>23</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuFile">
    <property name="title">
     <string>Fi&amp;le</string>
    </property>
    <addaction name="actionClear"/>
    <addaction name="actionExit"/>
   </widget>
   <widget class="QMenu" name="menuOptions">
    <property name="title">
     <string>D&amp;raw</string>
    </property>
    <widget class="QMenu" name="menu_Line">
     <property name="title">
      <string>&amp;Line</string>
     </property>
     <property name="icon">
      <iconset resource="opengl_resource.qrc">
       <normaloff>:/new/prefix1/images/icons8-design-48.png</normaloff>:/new/prefix1/images/icons8-design-48.png</iconset>
     </property>
     <addaction name="actionSimple_DDA"/>
     <addaction name="actionSymmetric_DDA"/>
     <addaction name="actionBresenham"/>
     <addaction name="actionMid_Point"/>
    </widget>
    <widget class="QMenu" name="menu_Circle">
     <property name="title">
      <string>&amp;Circle</string>
     </property>
     <property name="icon">
      <iconset resource="opengl_resource.qrc">
       <normaloff>:/new/prefix1/images/icons8-circle-48.png</normaloff>:/new/prefix1/images/icons8-circle-48.png</iconset>
     </property>
     <addaction name="actionCircle_Mid_Point"/>
    </widget>
    <addaction name="menu_Line"/>
    <addaction name="menu_Circle"/>
    <addaction name="actionEllipse"/>
    <addaction name="actionPolygon_2"/>
    <addaction name="actionSpecial_Polygon"/>
   </widget>
   <widget class="QMenu" name="menuProperties">
    <property name="title">
     <string>Properties</string>
    </property>
    <addaction name="actionColor"/>
    <addaction name="actionThickness"/>
    <addaction name="actionPattern"/>
    <addaction name="actionDetails"/>
   </widget>
   <widget class="QMenu" name="menuOptions_2">
    <property name="title">
     <string>Optio&amp;ns</string>
    </property>
    <widget class="QMenu" name="menuClip">
     <property name="title">
      <string>&amp;Clip</string>
     </property>
     <widget class="QMenu" name="menu_Line_2">
      <property name="title">
       <string>&amp;Line</string>
      </property>
      <addaction name="actionLiang"/>
      <addaction name="actionSutherland_Cohen"/>
     </widget>
     <widget class="QMenu" name="menuClip_Window">
      <property name="title">
       <string>&amp;Clip Window</string>
      </property>
      <addaction name="actionClip_Window"/>
      <addaction name="actionRemove_Clip_Window"/>
     </widget>
     <widget class="QMenu" name="menuWeiler_Atherton">
      <property name="title">
       <string>&amp;Weiler Atherton</string>
      </property>
      <addaction name="actionIntersection"/>
      <addaction name="actionUnion"/>
      <addaction name="actionDifference"/>
     </widget>
     <addaction name="menuClip_Window"/>
     <addaction name="menu_Line_2"/>
     <addaction name="menuWeiler_Atherton"/>
    </widget>
    <widget class="QMenu" name="menuFill">
     <property name="title">
      <string>&amp;Fill</string>
     </property>
     <addaction name="actionFill"/>
     <addaction name="actionFlood_Fill"/>
    </widget>
    <addaction name="actionSelect"/>
    <addaction name="actionUndo"/>
    <addaction name="actionRedo"/>
    <addaction name="actionDelete"/>
    <addaction name="actionGroup"/>
    <addaction name="menuClip"/>
    <addaction name="menuFill"/>
    <addaction name="actionRefresh"/>
   </widget>
   <widget class="QMenu" name="menuTransformations">
    <property name="title">
     <string>Transfor&amp;mations</string>
    </property>
    <addaction name="actionTranslate"/>
    <addaction name="actionScale"/>
    <addaction name="actionRotate"/>
   </widget>
   <addaction name="menuFile"/>
   <addaction name="menuOptions"/>
   <addaction name="menuProperties"/>
   <addaction name="menuOptions_2"/>
   <addaction name="menuTransformations"/>
  </widget>
  <widget class="QToolBar" name="mainToolBar">
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
   <addaction name="actionClear"/>
   <addaction name="separator"/>
   <addaction name="actionDelete"/>
   <addaction name="actionUndo"/>
   <addaction name="actionRedo"/>
   <addaction name="actionSelect"/>
   <addaction name="actionGroup"/>
   <addaction name="separator"/>
   <addaction name="actionColor"/>
   <addaction name="actionThickness"/>
   <addaction name="actionPattern"/>
   <addaction name="actionFlood_Fill"/>
   <addaction name="separator"/>
   <addaction name="actionSimple_DDA"/>
   <addaction name="actionCircle_Mid_Point"/>
   <addaction name="actionEllipse"/>
   <addaction name="actionPolygon_2"/>
   <addaction name="separator"/>
   <addaction name="actionIntersection"/>
   <addaction name="actionUnion"/>
   <addaction name="actionDifference"/>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
  <action name="actionClear">
   <property name="icon">
    <iconset resource="opengl_resource.qrc">
     <normaloff>:/new/prefix1/images/icons8-empty-trash-48.png</normaloff>:/new/prefix1/images/icons8-empty-trash-48.png</iconset>
   </property>
   <property name="text">
    <string>&amp;Clear</string>
   </property>
  </action>
  <action name="actionExit">
   <property name="icon">
    <iconset resource="opengl_resource.qrc">
     <normaloff>:/new/prefix1/images/icons8-close-window-48.png</normaloff>:/new/prefix1/images/icons8-close-window-48.png</iconset>
   </property>
   <property name="text">
    <string>&amp;Exit</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+W</string>
   </property>
  </action>
  <action name="actionColor">
   <property name="icon">
    <iconset resource="opengl_resource.qrc">
     <normaloff>:/new/prefix1/images/icons8-paint-palette-48.png</normaloff>:/new/prefix1/images/icons8-paint-palette-48.png</iconset>
   </property>
   <property name="text">
    <string>&amp;Color</string>
   </property>
  </action>
  <action name="actionThickness">
   <property name="icon">
    <iconset resource="opengl_resource.qrc">
     <normaloff>:/new/prefix1/images/icons8-line-width-48.png</normaloff>:/new/prefix1/images/icons8-line-width-48.png</iconset>
   </property>
   <property name="text">
    <string>&amp;Thickness</string>
   </property>
  </action>
  <action name="actionPattern">
   <property name="icon">
    <iconset resource="opengl_resource.qrc">
     <normaloff>:/new/prefix1/images/icons8-ellipse-64.png</normaloff>:/new/prefix1/images/icons8-ellipse-64.png</iconset>
   </property>
   <property name="text">
    <string>&amp;Pattern</string>
   </property>
  </action>
  <action name="actionSelect">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="icon">
    <iconset resource="opengl_resource.qrc">
     <normaloff>:/new/prefix1/images/icons8-cursor-48.png</normaloff>:/new/prefix1/images/icons8-cursor-48.png</iconset>
   </property>
   <property name="text">
    <string>&amp;Select</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+S</string>
   </property>
  </action>
  <action name="actionUndo">
   <property name="icon">
    <iconset resource="opengl_resource.qrc">
     <normaloff>:/new/prefix1/images/if_undo_100960.png</normaloff>:/new/prefix1/images/if_undo_100960.png</iconset>
   </property>
   <property name="text">
    <string>&amp;Undo</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Z</string>
   </property>
  </action>
  <action name="actionRedo">
   <property name="enabled">
    <bool>true</bool>
   </property>
   <property name="icon">
    <iconset resource="opengl_resource.qrc">
     <normaloff>:/new/prefix1/images/if_redo_101170.png</normaloff>:/new/prefix1/images/if_redo_101170.png</iconset>
   </property>
   <property name="text">
    <string>&amp;Redo</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+R</string>
   </property>
  </action>
  <action name="actionSimple_DDA">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="icon">
    <iconset resource="opengl_resource.qrc">
     <normaloff>:/new/prefix1/images/icons8-design-48.png</normaloff>:/new/prefix1/images/icons8-design-48.png</iconset>
   </property>
   <property name="text">
    <string>&amp;Simple DDA</string>
   </property>
   <property name="shortcut">
    <string>Alt+D</string>
   </property>
  </action>
  <action name="actionSymmetric_DDA">
   <property name="text">
    <string>Symmetric &amp;DDA</string>
   </property>
  </action>
  <action name="actionBresenham">
   <property name="text">
    <string>&amp;Bresenham</string>
   </property>
  </action>
  <action name="actionMid_Point">
   <property name="text">
    <string>&amp;Mid Point</string>
   </property>
  </action>
  <action name="actionCircle_Mid_Point">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="icon">
    <iconset resource="opengl_resource.qrc">
     <normaloff>:/new/prefix1/images/icons8-circle-48.png</normaloff>:/new/prefix1/images/icons8-circle-48.png</iconset>
   </property>
   <property name="text">
    <string>&amp;Circle Mid Point</string>
   </property>
   <property name="shortcut">
    <string>Alt+C</string>
   </property>
  </action>
  <action name="actionTranslate">
   <property name="text">
    <string>&amp;Translate</string>
   </property>
  </action>
  <action name="actionScale">
   <property name="text">
    <string>&amp;Scale</string>
   </property>
  </action>
  <action name="actionRotate">
   <property name="text">
    <string>&amp;Rotate</string>
   </property>
  </action>
  <action name="actiontesting">
   <property name="text">
    <string>t&amp;esting</string>
   </property>
  </action>
  <action name="actionDelete">
   <property name="icon">
    <iconset resource="opengl_resource.qrc">
     <normaloff>:/new/prefix1/images/icons8-cancel-64.png</normaloff>:/new/prefix1/images/icons8-cancel-64.png</iconset>
   </property>
   <property name="text">
    <string>&amp;Delete</string>
   </property>
   <property name="shortcut">
    <string>Del</string>
   </property>
  </action>
  <action name="actionRefresh">
   <property name="text">
    <string>R&amp;efresh</string>
   </property>
   <property name="shortcut">
    <string>F5</string>
   </property>
  </action>
  <action name="actionDetails">
   <property name="text">
    <string>&amp;Details</string>
   </property>
  </action>
  <action name="actionEllipse">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="icon">
    <iconset resource="opengl_resource.qrc">
     <normaloff>:/new/prefix1/images/icons8-oval-48.png</normaloff>:/new/prefix1/images/icons8-oval-48.png</iconset>
   </property>
   <property name="text">
    <string>&amp;Ellipse</string>
   </property>
  </action>
  <action name="actionFill">
   <property name="text">
    <string>&amp;Boundary Fill</string>
   </property>
  </action>
  <action name="actionClip_Window">
   <property name="text">
    <string>&amp;Draw</string>
   </property>
  </action>
  <action name="actionLiang">
   <property name="text">
    <string> &amp;Liang Barsky</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Alt+C</string>
   </property>
  </action>
  <action name="actionSutherland_Cohen">
   <property name="text">
    <string>&amp;Cohen Sutherland</string>
   </property>
  </action>
  <action name="actionRemove_Clip_Window">
   <property name="text">
    <string>&amp;Remove</string>
   </property>
  </action>
  <action name="actionFlood_Fill">
   <property name="icon">
    <iconset resource="opengl_resource.qrc">
     <normaloff>:/new/prefix1/images/icons8-fill-64.png</normaloff>:/new/prefix1/images/icons8-fill-64.png</iconset>
   </property>
   <property name="text">
    <string>&amp;Flood Fill</string>
   </property>
  </action>
  <action name="actionGroup">
   <property name="icon">
    <iconset resource="opengl_resource.qrc">
     <normaloff>:/new/prefix1/images/icons8-group-objects-48.png</normaloff>:/new/prefix1/images/icons8-group-objects-48.png</iconset>
   </property>
   <property name="text">
    <string>&amp;Group</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+G</string>
   </property>
  </action>
  <action name="actionPolygon_2">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="icon">
    <iconset resource="opengl_resource.qrc">
     <normaloff>:/new/prefix1/images/icons8-polygon-64.png</normaloff>:/new/prefix1/images/icons8-polygon-64.png</iconset>
   </property>
   <property name="text">
    <string>&amp;Polygon</string>
   </property>
  </action>
  <action name="actionSpecial_Polygon">
   <property name="text">
    <string>&amp;Special Polygon</string>
   </property>
  </action>
  <action name="actionIntersection">
   <property name="icon">
    <iconset resource="opengl_resource.qrc">
     <normaloff>:/new/prefix1/images/1-512.png</normaloff>:/new/prefix1/images/1-512.png</iconset>
   </property>
   <property name="text">
    <string>&amp;Intersection</string>
   </property>
  </action>
  <action name="actionUnion">
   <property name="icon">
    <iconset resource="opengl_resource.qrc">
     <normaloff>:/new/prefix1/images/10-512.png</normaloff>:/new/prefix1/images/10-512.png</iconset>
   </property>
   <property name="text">
    <string>&amp;Union</string>
   </property>
  </action>
  <action name="actionDifference">
   <property name="icon">
    <iconset resource="opengl_resource.qrc">
     <normaloff>:/new/prefix1/images/6-512.png</normaloff>:/new/prefix1/images/6-512.png</iconset>
   </property>
   <property name="text">
    <string>&amp;Difference</string>
   </property>
  </action>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <customwidgets>
  <customwidget>
   <class>gl_widget</class>
   <extends>QWidget</extends>
   <header>gl_widget.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources>
  <include location="opengl_resource.qrc"/>
 </resources>
 <connections/>
</ui>
